### PR TITLE
[Feature] Past audit comps leaderboard + claim rewards

### PR DESCRIPTION
**Pull request by fonstack**
### Summary by CodeRabbit

- New Feature: Added a new `ReleasePaymentSplitContract` class for releasing rewards from a split contract.
- Refactor: Updated the `parsePayouts` function to calculate `totalPaidOut` based on `hackerReward` and `hackerVestedReward`.
- Chore: Adjusted LogRocket initialization to only occur in non-local environments.
- New Feature: Introduced a `parseSeverityName` function for parsing and normalizing severity names.
- New Feature: Enhanced the Vault Details Page with a leaderboard section for audit competitions, allowing users to claim their bounty rewards if listed as winners.
- Refactor: Simplified the handling of finished audit competitions in the Private Audit Vaults Page by removing unnecessary dependencies.
- Refactor: Improved sorting of audit competition vaults and paid payouts from audits based on specific properties.

### CODE REVIEW

File: packages/web/src/hooks/subgraph/vaults/parser.ts [hunk 1]
```diff
@@ -31,12 +31,17 @@ export const parsePayouts = (payouts: IPayoutGraph[], chainId: number) => {
...
     isApproved: !!payout.approvedAt,
     isDismissed: !!payout.dismissedAt,
+    // totalPaidOut: !!payout.approvedAt
+    //   ? BigNumber.from(payout.committeeReward ?? "0")
+    //       .add(BigNumber.from(payout.hackerReward ?? "0"))
+    //       .add(BigNumber.from(payout.hackerVestedReward ?? "0"))
+    //       .add(BigNumber.from(payout.governanceHatReward ?? "0"))
+    //       .add(BigNumber.from(payout.hackerHatReward ?? "0"))
+    //       .toString()
+    //   : undefined,
```
Comment (shayzluf) [inline packages/web/src/hooks/subgraph/vaults/parser.ts:41]:
```suggestion
```

File: packages/web/src/contracts/write/ReleasePaymentSplit.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    const committeeCheckIn = useContractWrite({
+      mode: "recklesslyUnprepared",
+      address: splitContractAddress as `0x${string}` | undefined,
+      abi: HATPaymentSplitter_abi,
+      functionName: "release",
+      // chainId: vault.chainId,
+    });
+
+    return {
+      ...committeeCheckIn,
```
Comment (shayzluf) [inline packages/web/src/contracts/write/ReleasePaymentSplit.ts:25]:
```suggestion
      ... payoutRelease,
```

File: packages/web/src/contracts/write/ReleasePaymentSplit.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      send: async () => {
+        if (!vault.stakingToken || !address) return;
+
+        await switchNetworkAndValidate(chain!.id, vault?.chainId);
+
+        // [params]: token, account
+        return committeeCheckIn.write!({
```
Comment (shayzluf) [inline packages/web/src/contracts/write/ReleasePaymentSplit.ts:32]:
```suggestion
        return payoutRelease.write!({
```

**Review by shayzluf — CHANGES_REQUESTED** [review]
Will this leaderboard work with multiple payouts from the same vault ? 

hook error handling might be an issue.

File: packages/web/src/hooks/subgraph/vaults/parser.ts [hunk 4]
```diff
@@ -31,12 +31,17 @@ export const parsePayouts = (payouts: IPayoutGraph[], chainId: number) => {
...
     totalPaidOut: !!payout.approvedAt
-      ? BigNumber.from(payout.committeeReward ?? "0")
-          .add(BigNumber.from(payout.hackerReward ?? "0"))
+      ? BigNumber.from(payout.hackerReward ?? "0")
           .add(BigNumber.from(payout.hackerVestedReward ?? "0"))
-          .add(BigNumber.from(payout.governanceHatReward ?? "0"))
-          .add(BigNumber.from(payout.hackerHatReward ?? "0"))
```
[the remark this answers is not in the document]
Comment (fonstack) [reply packages/web/src/hooks/subgraph/vaults/parser.ts:39]: yes, but those two are the only things sent to the security researcher. That `totalPaidOut` property is the value paid to the hacker. @shayzluf

File: packages/web/src/contracts/write/ReleasePaymentSplit.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+          recklesslySetUnpreparedArgs: [vault?.stakingToken as `0x${string}`, "0x9834b17a29652e317202109C2e624FE0A32A1383"],
```
[the remark this answers is not in the document]
Comment (fonstack) [reply packages/web/src/contracts/write/ReleasePaymentSplit.ts:33]: pretty good catch @shayzluf. That's the address of the beneficiary, I put it for testing purposes and I forgot to revert it. Thanks

**Review by shayzluf — APPROVED** [review]

**Pull request merged**